test/lib/pmd/PmdEngine.test.ts

```diff
@@ -95,12 +96,101 @@ describe('Tests for BasePmdEngine and PmdEngine implementation', () => {
 
 			await (testPmdEngine as any).processStdOut(xml);
 			Sinon.assert.callCount(uxSpy, 3);
-			Sinon.assert.calledWith(uxSpy, 'warning-always', expectedConfigError);
-			Sinon.assert.calledWith(uxSpy, 'warning-always', expectedError);
-			Sinon.assert.calledWith(uxSpy, 'warning-always', expectedSuppressedViolation);
+			Sinon.assert.calledWith(uxSpy, EVENTS.WARNING_ALWAYS, expectedConfigError);
+			Sinon.assert.calledWith(uxSpy, EVENTS.WARNING_ALWAYS, expectedError);
+			Sinon.assert.calledWith(uxSpy, EVENTS.WARNING_ALWAYS, expectedSuppressedViolation);
 		});
 	});
 
+	describe('#filterSkippedRulesFromResults()', () => {
```

Comment:
Covered cases:
- Removing one violation from a result
- Removing all violations for a result and subsequently dropping the result altogether
- Confirming that the event fires as expected.